test/cases/fixes/incompleteTypes/reactTypes/reactPropsFromUses/expected.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+
+(function () {
+    interface FromUsesComponentProps {
+        returnsBoolean: () => unknown | (() => boolean);
+        takesStringCall: () => void | ((text: string) => void);
+        takesStringJsx: () => void | ((text: string) => string);
```

Comment:
These should really be replacements, not union types...

Reply:
I'm going to switch these to `unknown`s to start. There are more issues that I think will come down the pipeline about better resolution around functions and union types. In the meantime, this scenario is unlikely.